libs/serial/serial.ts

```diff
@@ -37,5 +37,55 @@ namespace serial {
         const prefix = name ? name + ":" : "";
         serial.writeLine(prefix + value);
     }
+    
+    /**
+     * Read a line of text from the serial port and return the buffer when the delimiter is met.
+     * @param delimiter text delimiter that separates each text chunk
+     */
+    //% help=serial/read-until
+    //% blockId=serial_readUntil block="serial|read until %delimiter=serial_delimiter_conv"
+    //% weight=19
+    export function readUntil(del: string): string {
```

Comment:
on a second though, you don't need the TypeScript implementatiuon fro this. Move the //% back to the C++ function and delete this function